drivers/modem/gsm_ppp.c

```diff
@@ -1097,6 +1102,26 @@ const struct gsm_ppp_modem_info *gsm_ppp_modem_info(const struct device *dev)
 	return &gsm->minfo;
 }
 
+static void gsm_mgmt_event_handler(struct net_mgmt_event_callback *cb,
+			  uint32_t mgmt_event, struct net_if *iface)
+{
+	if ((mgmt_event & NET_EVENT_IF_DOWN) != mgmt_event) {
+		return;
+	}
+
+	/* Right now we only support 1 GSM instance */
+	if (iface != gsm.iface) {
+		return;
+	}
+
+	if (mgmt_event == NET_EVENT_IF_DOWN) {
+		LOG_INF("GSM network interface down");
+		/* raise semaphore to indicate the interface is down */
+		k_sem_give(&gsm.sem_if_down);
+		return;
```

Comment:
Putting a deadline of 10 seconds on this call is a little too strict. When the terminate does not succeed, it will retry 10 times with a timeout of 1 second and will eventually terminate. Please extend to 15 seconds or more. The underlying source of the problem is interesting by itself.

Reply:
10 seconds is due to the limit of `task_wdt`, I think it is the longest timeout supported if I'm not wrong.

> it will retry 10 times with a timeout of 1 second and will eventually terminate.

where is this?

Reply:
subsys/net/l2/ppp/fsm.c line 211 starts the timeout of CONFIG_NET_L2_PPP_TIMEOUT msec default 3000
It trys for CONFIG_NET_L2_PPP_MAX_CONFIGURE_REQ_RETRANSMITS times, which is default 10

So the whole thing will wait max for 30 seconds. If you adjust these to be within 10 seconds you should be good.